Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -10321,11 +10321,13 @@ - (void)_didExitFullscreen
         }
     }
 
-    [item registerDataRepresentationForTypeIdentifier:utiType visibility:NSItemProviderRepresentationVisibilityAll loadHandler:[fileWrapper](void (^completionHandler)(NSData *, NSError *)) -> NSProgress * {
-        if (auto nsData = retainPtr([fileWrapper serializedRepresentation]))
-            completionHandler(nsData.get(), nil);
-        else
-            completionHandler(nil, [NSError errorWithDomain:WKErrorDomain code:WKErrorUnknown userInfo:nil]);
+    [item registerDataRepresentationForTypeIdentifier:utiType visibility:NSItemProviderRepresentationVisibilityAll loadHandler:[attachment](void (^completionHandler)(NSData *, NSError *)) -> NSProgress * {
+        attachment->doWithFileWrapper([&](NSFileWrapper *fileWrapper) {
+            if (auto nsData = RetainPtr { fileWrapper.serializedRepresentation })
```

Comment:
nit: Would this work?
`if (RetainPtr nsData = fileWrapper.serializedRepresentation)`

Reply:
Ah, so I originally tried exactly that, but got this error:

```
/Volumes/Data/worker/iOS-15-Simulator-Build-EWS/build/Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:10326:27: error: no viable constructor or deduction guide for deduction of template arguments of 'RetainPtr'
```